core/src/main/java/com/airbnb/aerosolve/core/util/SupportVector.java

```diff
@@ -0,0 +1,90 @@
+package com.airbnb.aerosolve.core.util;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Random;
+
+import com.airbnb.aerosolve.core.FunctionForm;
+import com.airbnb.aerosolve.core.ModelRecord;
+
+public class SupportVector implements Serializable {
```

Comment:
I think abstract function was made too specific, with things like Linfinity cap, updating at X ... support vector takes in vectors too not scalars so I don't think I will do this change since it is a vector function that returns a scalar instead of a scalar function taking a scalar.